core/include/scipp/core/element/geometric_operations.h

```diff
@@ -16,7 +16,68 @@ namespace scipp::core {
 
 namespace element {
 
-namespace geometry {
+    constexpr auto less = overloaded{
+        arg_list<double, float, int64_t, int32_t>,
+        transform_flags::expect_no_variance_arg<0>,
+        transform_flags::expect_no_variance_arg<1>,
+        [](const auto &x, const auto &y) {return x < y; },
+        [](const units::Unit &x, const units::Unit &y) {
+            expect::equals(x, y);
+            return units::dimensionless;
+        }
+    };
+    constexpr auto greater = overloaded{
+        arg_list<double, float, int64_t, int32_t>,
+        transform_flags::expect_no_variance_arg<0>,
+        transform_flags::expect_no_variance_arg<1>,
+        [](const auto &x, const auto &y) { return x > y; },
+        [](const units::Unit &x, const units::Unit &y) {
+            expect::equals(x, y);
+            return units::dimensionless;
+        }
+    };
```

Comment:
Looking at this pattern, can you shorten things as follows?

```python
    constexpr auto comparison = overloaded{
        arg_list<double, float, int64_t, int32_t>,
        transform_flags::expect_no_variance_arg<0>,
        transform_flags::expect_no_variance_arg<1>,
        [](const units::Unit &x, const units::Unit &y) {
            expect::equals(x, y);
            return units::dimensionless;
        }
    };
    constexpr auto greater = overloaded{
        comparison,
        [](const auto &x, const auto &y) { return x > y; }
    };
    // same for less, ...
```

This should also allow for less duplicaiotn in the tests, since you can test `comparison` independently.

Reply:
VERY good comment. thanks.